### PR TITLE
[Skunkworks] Experimental auditing controller

### PR DESCRIPTION
**Pull request by josvazg**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: pkg/controller/atlasauditing/controller.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	if customresource.ReconciliationShouldBeSkipped(auditing) {
+		r.Log.Infow(fmt.Sprintf("-> Skipping AtlasAuditing reconciliation as annotation %s=%s", customresource.ReconciliationPolicyAnnotation, customresource.ReconciliationPolicySkip), "spec", auditing.Spec)
+		return workflow.OK().ReconcileResult(), nil
+	}
+
+	// TODO: Fix! we no longer want the status to flip&flap but be updated only at the end
+	//       not sure what we chose to replace this with
+	// workflowCtx := customresource.MarkReconciliationStarted(r.Client, databaseUser, log, ctx)
```
Comment (s-urbaniak) [inline pkg/controller/atlasauditing/controller.go:50]: see https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1595 with a replacement, we have everything in master already.

**Comment by josvazg** [comment]
Closing in favor of #1605

**Comment by josvazg** [comment]
Actually this is low priority, but still valid. It does need to be rebased after #1605 though

**Comment by josvazg** [comment]
Seems it is time to close this one for good WDYT @s-urbaniak ? We do not want to pursue this strategy right?

**Comment by josvazg** [comment]
Will close, we can still check the code for tips when or if we need to get back to this. But as it will not happen soon, direct rebasing will not make much sense.

**Pull request closed without merging**